services/121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/services/cooperative-bank-of-oromia-api.service.ts

```diff
@@ -0,0 +1,205 @@
+    if (
+      env.MOCK_COOPERATIVE_BANK_OF_OROMIA ||
+      !env.COOPERATIVE_BANK_OF_OROMIA_API_URL
+    ) {
+      this.cooperativeBankOfOromiaTransferURL = new URL(
+        `api/fsp/cooperative-bank-of-oromia/${transferUrlPart}`,
+        env.MOCK_SERVICE_URL,
+      );
+    } else {
+      this.cooperativeBankOfOromiaTransferURL = new URL(
+        transferUrlPart,
+        env.COOPERATIVE_BANK_OF_OROMIA_API_URL,
+      );
+    }
+
+    // Set auth base url; this is a different api path
+    let authUrlBase: URL;
+    if (
+      env.MOCK_COOPERATIVE_BANK_OF_OROMIA ||
+      !env.COOPERATIVE_BANK_OF_OROMIA_AUTH_URL
+    ) {
+      authUrlBase = new URL(
+        'api/fsp/cooperative-bank-of-oromia/',
+        env.MOCK_SERVICE_URL,
+      );
+    } else {
+      authUrlBase = new URL(env.COOPERATIVE_BANK_OF_OROMIA_AUTH_URL);
+    }
+    const urlPart = 'oauth2/token';
+    this.cooperativeBankOfOromiaAuthenticateURL = new URL(urlPart, authUrlBase);
+  }
+
+  public async initiateTransfer({
+    cooperativeBankOfOromiaMessageId,
+    recipientCreditAccountNumber,
+    debitAccountNumber,
+    amount,
+  }: {
+    cooperativeBankOfOromiaMessageId: string;
+    recipientCreditAccountNumber: string;
+    debitAccountNumber: string;
+    amount: number;
+  }): Promise<{
+    result: CooperativeBankOfOromiaTransferResultEnum;
+    message?: string;
+  }> {
+    let tokenSet: TokenSet;
+    try {
+      tokenSet = await this.authenticate();
+    } catch (error) {
+      if (error instanceof CooperativeBankOfOromiaApiError) {
+        return {
+          result: CooperativeBankOfOromiaTransferResultEnum.fail,
+          message: error.message,
+        };
+      } else {
+        throw error;
+      }
+    }
+
+    const headers = this.addAuthHeaders({
+      headers: new Headers({
+        Accept: '*/*',
+        'Content-Type': 'application/json',
+      }),
+      tokenSet,
+    });
+
+    const payload: CooperativeBankOfOromiaApiTransferRequestBodyDto =
+      this.cooperativeBankOfOromiaApiHelperService.buildTransferPayload({
+        cooperativeBankOfOromiaMessageId,
+        recipientCreditAccountNumber,
+        debitAccountNumber,
+        amount,
+      });
+
+    let response: AxiosResponse<CooperativeBankOfOromiaApiPaymentResponseBodyDto>;
+
+    try {
+      response = await this.httpService.post<
+        AxiosResponse<CooperativeBankOfOromiaApiPaymentResponseBodyDto>
+      >(
+        this.cooperativeBankOfOromiaTransferURL.href,
+        payload,
+        this.httpService.headersToPojo(headers),
+      );
+    } catch (error) {
+      return {
+        result: CooperativeBankOfOromiaTransferResultEnum.fail,
+        message: `Transfer failed: ${error.message}`,
+      };
+    }
+
+    // Check if the response indicates success
+    return this.cooperativeBankOfOromiaApiHelperService.handleTransferResponse(
+      response.data,
+    );
+  }
+
+  private addAuthHeaders({
```

Comment:
should this go to api.helper? Or is a separate method at all needed actually, given it's only used once and adds more lines than it saves?